# 5장 과제

## Problem 1

In [2]:
import numpy as np

# 학급 학생들의 키를 담은 numpy 배열을 생성 
height = np.array([167.3, 174.3, 175.0, 177.6, 160.5, 161.0, 177.4])

# 키의 평균, 표준편차, 중앙값을 계산하여 출력
print('키의 평균 : %.2fcm' % height.mean())
print('키의 표준편차 : %.2fcm' % height.std())
print('키의 중앙값 : %.2fcm' % np.median(height))

키의 평균 : 170.44cm
키의 표준편차 : 6.90cm
키의 중앙값 : 174.30cm


## Problem 2

In [52]:
from scipy.stats import norm

# seed를 설정한다
np.random.seed(5)

# Xi ~ N(0, 1)
# mu1, mu2, mu3 는 각각 표본의 크기 100, 10, 50일 때의 표본평균을 의미한다
mu1, mu2, mu3 = [], [], []

for i in range(100):
    # 크기 100의 표본을 추출한다.
    sample = norm.rvs(loc=0, scale=1, size=100)
    # 표본의 크기 100, 10, 50의 표본평균을 각각 mu1, mu2, mu3에 저장한다
    mu1.append(np.mean(sample))
    mu2.append(np.mean(sample[:10]))
    mu3.append(np.mean(sample[:50]))

# mu1, mu2, mu3의 평균을 계산한다
print('mean of mu1: %.5f' %np.mean(mu1))
print('mean of mu2: %.5f' %np.mean(mu2))
print('mean of mu3: %.5f' %np.mean(mu3))

# mu1, mu2, mu3의 표준편차를 계산한다
print('\nstd of mu1: %.3f' %np.std(mu1))
print('std of mu2: %.3f' %np.std(mu2))
print('std of mu3: %.3f' %np.std(mu3))

# mu1, mu2, mu3의 분산을 계산한다
print('\nvar of mu1: %.3f' %np.var(mu1))
print('var of mu2: %.3f' %np.var(mu2))
print('var of mu3: %.3f' %np.var(mu3))


mean of mu1: -0.00029
mean of mu2: -0.00345
mean of mu3: -0.02773

std of mu1: 0.113
std of mu2: 0.307
std of mu3: 0.156

var of mu1: 0.013
var of mu2: 0.094
var of mu3: 0.024


추정량의 평균의 경우 mu1이 0에 가장 가까웠으며 다음으로 mu2, mu3의 순서였다.  
추정량의 분산의 경우 mu1이 가장 작았으며 그 다음으로 mu3, mu2의 순서로 나타났다.  
따라서 표본의 크기가 클수록 추정량의 분산이 작아지는 경향을 확인할 수 있다.

(추정량의 평균에서 mu2와 mu3의 비교가 예상과 다르게 나타난 것은 우연적인 효과이다.  
random seed를 5가 아닌 값으로 설정한 경우 mu3의 값이 mu2에 비해 0에 더 가까운 것이 일반적이었다. 표본의 크기가 클수록 추정량의 평균이 모평균에 더 가깝게 나타나는 경향이 있다는 뜻이다.)

## Problem 3

표본의 크기: 9  
모분산: 16  
95% 신뢰구간  

In [11]:
# 표본의 크기, 모표준편차, alpha값을 저장
sample_num = 9
sigma = 4
alpha = 0.05

# 크기 9의 표본을 나타내는 리스트 
sample_lst = np.array([54.0 , 54.3, 51.0, 52.2, 56.6, 53.7, 57.0, 55.9, 56.1])

# 표본 평균을 계산한다
sample_mean = sample_lst.mean()
print('표본 평균: %.3f시간' %sample_mean.mean())

# n은 N(0, 1)의 표준정규분포
n = norm(loc=0, scale=1)

# z_0.025의 값을 계산
z_val = n.ppf(1-alpha/2)
print('z_0.025의 값 %.3f' %z_val)

표본 평균: 54.533시간
z_0.025의 값 1.960


In [41]:
# 95% 신뢰구간을 계산한다
upper = sample_mean + z_val * (sigma / np.sqrt(sample_num))
lower = sample_mean - z_val * (sigma / np.sqrt(sample_num))
print('95%% 신뢰구간: [%.3f, %.3f]' %(lower, upper))

95% 신뢰구간: [15.045, 11.755]


## Problem 4

도시의 온도의 모평균을 $\mu$라고 하자.  

H0: $\mu$ = 15  
H1: $\mu$ < 15

유의수준은 5%로 설정한다.
표본의 크기는 4이다.  
도시의 온도 X ~ N($\mu$, 9)  
$\overline{X}$ ~ N($\mu$, 9/4)

In [39]:
sample_mean = 13.4
sigma = 3
mu = 15

# 검정통계량은 N(0, 1)을 따른다.
# 그 값은 아래와 같다.
test_stat = (sample_mean - mu) / (sigma / 2)
print('검정통계량 Z = (표본평균 - 모평균) / (모표준편차 / sqrt(표본 크기)) ~ N(0, 1)')
print('\n검정통계량의 값: %.3f' %test_stat)

검정통계량 Z = (표본평균 - 모평균) / (모표준편차 / sqrt(표본 크기)) ~ N(0, 1)

검정통계량의 값: -1.067


In [40]:
from scipy.stats import norm

# 유의수준 5%에서의 기각역을 계산하자
alpha = 0.05
n = norm(loc=0, scale=1)
z_val = norm.ppf(alpha)

# 기각역을 출력한다
print('기각역은 Z < %.3f 이다.' %z_val)
print('단, Z는 N(0, 1)을 따르는 검정통계량이다.')

기각역은 Z < -1.645 이다.
단, Z는 N(0, 1)을 따르는 검정통계량이다.


검정통계량의 값인 -1.067은 기각역 Z < -1.645에 포함되지 않는다.  
따라서 귀무가설 H0는 기각하지 않는다.  
즉 유의수준 5%에서 도시의 온도가 15도보다 낮다고 볼 충분한 통계적 근거가 없다.

## Problem 5

제품의 무게의 모평균을 $\mu$라 하자.  

H0: $\mu$ = 500  
H1: $\mu$ > 500

$\sigma$ = 4  

제품의 무게 $X$ ~ N($\mu$, 16)  
$\overline{X}$ ~ N($\mu$, 16/9)

In [38]:
# 크기 9인 표본은 아래와 같이 주어졌다.
sample_lst = np.array([502, 510, 497, 495, 508, 511, 503, 500, 496])

# 표본평균을 계산한다
sample_mean = sample_lst.mean()
print('표본평균: %.3f' %sample_mean)

# 검정통계량 Z의 값을 계산한다.
print('검정통계량 Z = (표본평균 - 모평균) / (모표준편차 / sqrt(표본 크기)) ~ N(0, 1)')
test_stat = (sample_mean - 500) / (4 / 3)
print('\n검정통계량의 값: %.3f' %test_stat)

표본평균: 502.444
검정통계량 Z = (표본평균 - 모평균) / (모표준편차 / sqrt(표본 크기)) ~ N(0, 1)

검정통계량의 값: 1.833


In [37]:
from scipy.stats import norm

# 1-p의 값을 계산한다.
cdf_val = norm.cdf(test_stat, loc=0, scale=1)

# 유의확률 p의 값을 계산하여 출력한다.
p_val = 1 - cdf_val
print('유의확률: %.3f' %p_val)

유의확률: 0.033
